File: component/src/main/java/com/mackwu/component/SecondActivity.java
package com.mackwu.component;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;

import com.mackwu.xmvc.BaseActivity;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

import butterknife.OnClick;

/**
 * ===================================================
 * Created by MackWu on 2020/11/4 15:09
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
public class SecondActivity extends BaseActivity {

    private ExecutorService mSingleInstallThread = Executors.newSingleThreadExecutor();

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public void initView(@Nullable Bundle savedInstanceState) {

    }

    @OnClick(R.id.btn_test)
    public void onBtnTestClicked() {
        mSingleInstallThread.execute(() -> {
            Log.d("TAG", "XXX");
        });
    }

}

File: http/src/main/java/com/mackwu/http/SecondActivity.kt
package com.mackwu.http

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.mackwu.http.okhttp_retrofit_rxjava.ApiHelper
import kotlinx.android.synthetic.main.activity_main.*

/**
 * ===================================================
 * Created by MackWu on 2020/1/6 19:05
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        btn_test.setOnClickListener { ApiHelper.getIp {} }
    }
}
